### PR TITLE
[MM-37468] Update tutorial dot

### PR DESCRIPTION
**Pull request by neallred**
#### Summary
Change tutorial dot icon and update the first for tutorial tips' wording as per the design doc: https://docs.google.com/document/d/1vn7RIiP6pbbaW8MpH1w_vnoBW8gMBepYu9X1ridh96k/edit#

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-37468
https://mattermost.atlassian.net/browse/MM-37467
https://mattermost.atlassian.net/browse/MM-37469
https://mattermost.atlassian.net/browse/MM-37470
https://mattermost.atlassian.net/browse/MM-37471
https://mattermost.atlassian.net/browse/MM-37486

Since these tickets are smaller and all highly related, I batched them. They all correspond to existing tutorial dots. A subsequent effort will add the remaining tutorial dots we want to add.

#### Related Pull Requests

None

#### Screenshots

https://user-images.githubusercontent.com/13738432/128062784-02fc7150-02c4-4b60-a862-670b401b6ad7.mp4

#### Release Note

```release-note
* Update tutorial icon
* Change text content in tutorial tips
```

### CODE REVIEW

**Review by esethna — CHANGES_REQUESTED** [review]
Thanks @neallred, few things:

1. 
image

2. I updated the copy slightly, see the doc. Also:
image

3. Updated the copy for the third dot to make it shorter https://docs.google.com/document/d/1vn7RIiP6pbbaW8MpH1w_vnoBW8gMBepYu9X1ridh96k/edit#bookmark=id.faw7ptb2wbt

4. Move the dot on the hamburger to the top right of the icon (like the plus button)
image

**Review by esethna — APPROVED** [review]
Looks good pending the additional pop-overs FIVE and SIX in a separate PR. 

This PR will also need to be tested with the global header on

cc// @stevemudie 

Approving on behalf of UX as well

File: components/sidebar/add_channel_dropdown.tsx [hunk 1]
```diff
@@ -105,19 +110,32 @@ class AddChannelDropdown extends React.PureComponent<Props, State> {
...
                     placement='top'
                     overlay={tooltip}
                 >
-                    <button
-                        className='AddChannelDropdown_dropdownButton'
-                        aria-label={intl.formatMessage({id: 'sidebar_left.add_channel_dropdown.dropdownAriaLabel', defaultMessage: 'Add Channel Dropdown'})}
-                    >
-                        <i className='icon-plus'/>
-                    </button>
+                    <React.Fragment>
```
Comment (pablovelezvidal) [inline components/sidebar/add_channel_dropdown.tsx:130]: we can replace `<React.Fragment>` for its shorter syntax `<></>`

File: components/legacy_sidebar/channel_tutorial_tip.jsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
```
Comment (pablovelezvidal) [inline components/legacy_sidebar/channel_tutorial_tip.jsx:1]: since we are touching this file, maybe we want to try to turn it to .TSX

File: components/sidebar/index.ts [hunk 3]
```diff
@@ -12,7 +12,6 @@ import {getBool} from 'mattermost-redux/selectors/entities/preferences';
 import {haveICurrentChannelPermission} from 'mattermost-redux/selectors/entities/roles';
 import {getCurrentTeam} from 'mattermost-redux/selectors/entities/teams';
 import {GenericAction} from 'mattermost-redux/types/actions';
-
```
Comment (pablovelezvidal) [inline components/sidebar/index.ts:15]: maybe we want this line for separating imports

File: components/widgets/pulsating_dot/index.ts [hunk 4]
```diff
@@ -0,0 +1,4 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
```
Comment (pablovelezvidal) [inline components/widgets/pulsating_dot/index.ts:1]: not sure if we want to keep this index file, I don't know if there is another benefit different than a shorter import statement.

Comment (neallred) [reply components/widgets/pulsating_dot/index.ts:1]: 0/5 preference on keeping it for shorter import path and conformity with an existing pattern throughout the app (found more than 20 other instances of index files that only re-exported without doing anything else).

**Comment by stevemudie** [comment]
@neallred This worked as expected until it came to the dot hovering partially behind the red dot of the ☰ menu.

Each time I clicked on it the tip would only expand the menu.
https://user-images.githubusercontent.com/7083912/128224619-9c04134b-5381-4db7-950a-1e48c524f025.mp4

**Comment by neallred** [comment]
Interesting. I'll remove the red dot there and in the marketplace menu since that is another ticket (https://mattermost.atlassian.net/browse/MM-37486) and is the easiest fix.

**Comment by AGMETEOR** [comment]
Hi @esethna do we still need to maintain code in the legacy sidebar?

**Comment by esethna** [comment]
@AGMETEOR no, since it's being deprecated in in v6.0 and cannot be enabled in cloud

**Comment by neallred** [comment]
Thanks for the clarifications. @AGMETEOR since we don't need to continue maintaining it, I can move `channel_tutorial_tip` to the `components/sidebar` folder.

File: components/legacy_sidebar/sidebar_channel/sidebar_channel.jsx [hunk 5]
```diff
@@ -12,7 +12,7 @@ import {intlShape} from 'utils/react_intl';
 import {trackEvent} from 'actions/telemetry_actions.jsx';
 import * as GlobalActions from 'actions/global_actions';
 import SidebarChannelButtonOrLink from '../sidebar_channel_button_or_link/sidebar_channel_button_or_link.jsx';
-import SidebarTutorialTip from '../sidebar_tutorial_tip.jsx';
```
Comment (AGMETEOR) [inline components/legacy_sidebar/sidebar_channel/sidebar_channel.jsx:15]: @neallred from a first pass, everything looks great. I think we should revert the changes made in the legacy sidebar code

**Comment by AGMETEOR** [comment]
> Thanks for the clarifications. @AGMETEOR since we don't need to continue maintaining it, I can move `channel_tutorial_tip` to the `components/sidebar` folder.

Yes.

**Comment by stevemudie** [comment]
@neallred I effectively removed the red dot by visiting the plugin marketplace and even though it's now gone only the ☰ menu  responds to clicks.

Is it a placement issue that's causing it since it's so tight to the ☰?

**Comment by neallred** [comment]
@stevemudie Its supposed to be so that clicking either the hamburger or the dot will bring up the tutorial, which I remember working on my machine. I will double check it on my machine and let you know what I find.

File: components/legacy_sidebar/sidebar_channel/sidebar_channel.jsx (hunk 5 above)
Comment (neallred) [reply components/legacy_sidebar/sidebar_channel/sidebar_channel.jsx:15]: I can do that.

**Comment by neallred** [comment]
@pablovelezvidal @AGMETEOR rerequesting review because of some substantive code changes. I made the changes because I needed to remove the red dot for admins that haven't visited the marketplace; it was interfering with the pulsating green dot. But to do that, I needed to change up another legacy_sidebar component, and so rather than do that I copied it to sidebar and converted it to typescript. There are a handful of changes to types in unrelated files to make everything typecheck.

**Comment by neallred** [comment]
@stevemudie I forgot to make the same changes I made to the "plus" button to the hamburger, but it should work as expected now :hamburger: !

**Review by stevemudie — APPROVED** [review]
Tested and passed! 

All of the tips are obvious and clear and worked as expected.

File: components/sidebar/add_channel_dropdown.tsx [hunk 6]
```diff
@@ -105,19 +110,32 @@ class AddChannelDropdown extends React.PureComponent<Props, State> {
...
                     placement='top'
                     overlay={tooltip}
                 >
-                    <button
-                        className='AddChannelDropdown_dropdownButton'
-                        aria-label={intl.formatMessage({id: 'sidebar_left.add_channel_dropdown.dropdownAriaLabel', defaultMessage: 'Add Channel Dropdown'})}
-                    >
-                        <i className='icon-plus'/>
-                    </button>
+                    <>
```
Comment (AGMETEOR) [inline components/sidebar/add_channel_dropdown.tsx:130]: nit: why the fragment tag?

File: components/sidebar/add_channel_tutorial_tip.tsx [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+        }
+
+        let offTopicDisplayName = Constants.OFFTOPIC_CHANNEL_UI_NAME;
+        if (this.props.offTopicDisplayName) {
+            offTopicDisplayName = this.props.offTopicDisplayName;
+        }
+
+        const screens = [];
+
+        screens.push(
```
Comment (AGMETEOR) [inline components/sidebar/add_channel_tutorial_tip.tsx:31]:
```suggestion
        const screens = (
```

File: components/sidebar/add_channel_tutorial_tip.tsx [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+        if (this.props.townSquareDisplayName) {
+            townSquareDisplayName = this.props.townSquareDisplayName;
+        }
+
+        let offTopicDisplayName = Constants.OFFTOPIC_CHANNEL_UI_NAME;
+        if (this.props.offTopicDisplayName) {
+            offTopicDisplayName = this.props.offTopicDisplayName;
+        }
+
+        const screens = [];
```
Comment (AGMETEOR) [inline components/sidebar/add_channel_tutorial_tip.tsx:29]:
```suggestion
```

File: components/sidebar/channel_tutorial_tip.tsx [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+
+export default class ChannelTutorialTip extends React.PureComponent<Props> {
+    componentDidMount() {
+        if (this.props.townSquareDisplayName || this.props.offTopicDisplayName) {
+            this.props.openLhs();
+        }
+    }
+
+    render = () => {
+        const screens = [];
```
Comment (AGMETEOR) [inline components/sidebar/channel_tutorial_tip.tsx:25]:
```suggestion
```

File: components/sidebar/channel_tutorial_tip.tsx [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+
+        screens.push(
```
Comment (AGMETEOR) [inline components/sidebar/channel_tutorial_tip.tsx:27]:
```suggestion
        const screens = (
```

File: components/sidebar/sidebar_header/dropdown/index.ts [hunk 11]
```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
```
Comment (AGMETEOR) [inline components/sidebar/sidebar_header/dropdown/index.ts:1]: I understand we want to migrate this component from the legacy folder to here. But, I would think it's better to do that in another ticket. We could create other tickets to do some of that cleaning up. We want to reduce the scope of this PR to only the ticket at hand in order to avoid any regressions we didn't see coming.

File: components/widgets/pulsating_dot/index.ts (hunk 4 above)
Comment (AGMETEOR) [reply components/widgets/pulsating_dot/index.ts:1]: Since it's already in the folder `pulsating_dot`, renaming it `index.ts` will make it such that there's no need for this extra file.

File: components/widgets/pulsating_dot/pulsating_dot.tsx [hunk 12]
```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
```
Comment (AGMETEOR) [inline components/widgets/pulsating_dot/pulsating_dot.tsx:1]: We can rename this file to `index.tsx` to avoid the extra `index` file in this folder that just imports and exports

**Review by AGMETEOR — CHANGES_REQUESTED** [review]

**Comment by AGMETEOR** [comment]
> @pablovelezvidal @AGMETEOR rerequesting review because of some substantive code changes. I made the changes because I needed to remove the red dot for admins that haven't visited the marketplace; it was interfering with the pulsating green dot. But to do that, I needed to change up another legacy_sidebar component, and so rather than do that I copied it to sidebar and converted it to typescript. There are a handful of changes to types in unrelated files to make everything typecheck.

4/5 we don't make changes that're not pertinent to the ticket

**Comment by pablovelezvidal** [comment]
> > @pablovelezvidal @AGMETEOR rerequesting review because of some substantive code changes. I made the changes because I needed to remove the red dot for admins that haven't visited the marketplace; it was interfering with the pulsating green dot. But to do that, I needed to change up another legacy_sidebar component, and so rather than do that I copied it to sidebar and converted it to typescript. There are a handful of changes to types in unrelated files to make everything typecheck.
> 
> 4/5 we don't make changes that're not pertinent to the ticket

I suggested that the file @neallred  was touching and moving place  should actually be migrated to TS, but I agree with Allan that we should keep the scope of the changes to just modify the files involved/required by the ticket. Then, if you want and have the time, could push another PR adding the migrations to TS and the code quality improvements.

**Comment by neallred** [comment]
As I understand it, there's two main sets of changes causing concern about too much scope.

1) converting `modal.js` to typescript. It is a widely used file, and converting it led to many unrelated changes to make things type check. I'm down for reverting those changes in this PR.

2) Duplicating and then modifying legacy sidebar components instead of changing them in place. I don't think there's a risk of regression here; if anything, there's less chance of impacting the still existing legacy sidebar.  I think we should keep these changes. They've already been reviewed and tested to work as expected; reverting them now would be extra work in the form of extra PRs and more rounds of testing. But I can definitely make my PRs smaller in the future.

File: components/main_menu/index.jsx [hunk 13]
```diff
@@ -7,7 +7,6 @@ import {bindActionCreators} from 'redux';
 import {
     getConfig,
     getLicense,
-    getFirstAdminVisitMarketplaceStatus,
```
[the remark this answers is not in the document]
Comment (neallred) [reply components/main_menu/index.jsx:10]: All the firstAdminVisitMarketplaceStatus/getFirstAdminVisitMarketplaceStatus changes are intentional and are part of implementing https://mattermost.atlassian.net/browse/MM-37486

File: components/sidebar/add_channel_dropdown.tsx (hunk 6 above)
Comment (neallred) [reply components/sidebar/add_channel_dropdown.tsx:130]: OverlayTrigger only allows a single react child. I got build errors without the fragment.

**Review by AGMETEOR — APPROVED** [review]
Great work on this @neallred . Everything looks good to me now.

**Review by stevemudie — APPROVED** [review]
Red dot no longer present in ☰ menu and all 4 tips appear as expected. Looks good!

**Review by pablovelezvidal — APPROVED** [review]
Great work @neallred ! LGTM! :shipit:

**Comment by neallred** [comment]
Thanks all!
